docs/specs/README.md

```diff
@@ -0,0 +1,64 @@
+# Guidelines for Feature Request(s)
+
+This document covers, how to contribute new feature(s) or significant design changes to BigchainDB. Since, contribution
+of such functionality involves design discussion and might take multiple releases to complete,
+we expect the contributors to follow certain guidelines before proceeding with the implementation.
+
+
+## Classification
+How to classify your contribution as a feature and determine if it needs design discussion. 
+
+Here are some baseline criterion that can help you classify your idea, as a feature:
+
+- Requires significant effort/changes to the current BigchainDB implementation and it is not self
+ contained (e.g. API changes,  introducing a new component, takes more than 20 Wrigleys
+ (Wrigley is a `man day` @ BigchainDB))
+- Impacts the clients and drivers.
+- Can we write a blog post about it?
+- Requires collaboration with multiple teams in terms of development, operations and testing.
+- Impacts documentation significantly.
+
+It is not a feature if:
+
+- It is straightforward enough and might not need a design discussion.
+  - Test case fixes
+  - Fixing typos/grammar in documentation.
+  - Refactoring code
+  - Adding error or log messages.
+
+## Classified as a Feature?
+Create a new issue on our [GitHub repository](https://github.com/bigchaindb/bigchaindb/issues/new) with a
+`Feature-Request` label, once you have classified your proposal as a feature and:
+
+- Scratched the surface.
+  - Discussed the proposal in the community e.g. on [our Gitter channel](https://gitter.im/bigchaindb/bigchaindb),
+  Or [Twitter](https://twitter.com/BigchainDB) and saw interest.
+- *Optional:* Have a working prototype.
+- Worked out the resources and people, who can contribute towards this feature.
+
+## Submit Spec
+After creating an issue with an appropriate problem statement and details, submit a design specification document
```

Comment:
This first line should let them know, right away, that they have to create a _pull request_ containing a design specification document.

I worry that some people might get upset by all the busywork at this point and leave. Maybe we could let them draft a feature request in their issue, and then _we_ could do all the work of turning it into a formal design document with a pull request, design review, etc.?

There's another problem too. Some feature requests will be requests for changes _in the IPDB Protocol_ or to drivers, which don't even live in this repository. Those requests will have to be moved, which is fine, just so long as we remain mindful of what lives where.

We can keep them updated of what we're doing by commenting in their original issue.

Reply:
for IPDB related things, maybe we can introduce a similar structure in those repositories as well or just keep track internally.

I think, if someone is proposing a significant changes, I think they will be willing to go over this process. I think this is much less tedious than the ones going around in the community.

Reply:
But I will take care of mentioning that they need to create a PR>

Reply:
I guess we can improve and simplify this process over time.